File: Ejercicios POO/PHP/Herencia y Composicion/Autos.php
<?php

// Written By Ismael Heredia in the year 2016

class Auto
{
    
    private $marca;
    private $numero_serie;
    
    public function __construct()
    {
        $this->marca   = "";
        $this->numero_serie = "";
    }
    
    public static function CreateAuto($marca,$numero_serie)
    {
        $instance                 = new self();
        $instance->marca   = $marca;
        $instance->numero_serie = $numero_serie;
        return $instance;
    }
    
    public function getMarca()
    {
        return $this->marca;
    }
    
    public function setMarca($marca)
    {
        $this->marca = $marca;
    }
    
    public function getNumero_serie()
    {
        return $this->numero_serie;
    }
    
    public function setNumero_serie($numero_serie)
    {
        $this->numero_serie = $numero_serie;
    }
        
    public function __destruct()
    {
        $this->marca   = "";
        $this->numero_serie = "";
    }
    
    public function DatosAuto()
    {
        return "-- Datos del auto --" . "\n\n"
            . "[+] Marca : " . $this->marca . "\n"
            . "[+] Numero de serie : " . $this->numero_serie;
    }
    
}

?>

File: Ejercicios POO/Python/Herencia y Composicion/Autos.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Written By Ismael Heredia in the year 2016

class Auto(object):

   def __init__(self, marca="", numero_serie=""):
       self.__marca = marca
       self.__numero_serie = numero_serie
              
   def getMarca(self):
       return self.__marca

   def setMarca(self, marca = ""):
       self.__marca = marca

   def getNumero_serie(self):
       return self.__numero_serie

   def setNumero_serie(self, numero_serie = ""):
       self.__numero_serie = numero_serie
           
   def DatosAuto(self):
       return "-- Datos del auto --\n\n[+] Marca : %s\n[+] Numero serie : %s" % (self.__marca,self.__numero_serie)
          
   def destroy(self):
       self.__marca = None
       self.__numero_serie = None

